FuelSDK-CSharp/ETClient.cs

```diff
@@ -338,7 +346,12 @@ private void RefreshTokenWithOauth2(bool force = false)
 
             using (var streamWriter = new StreamWriter(request.GetRequestStream()))
             {
-                string json = string.Format("{{\"client_id\": \"{0}\", \"client_secret\": \"{1}\", \"grant_type\": \"client_credentials\"}}", configSection.ClientId, configSection.ClientSecret);
+                string json = string.Format("{{\"client_id\": \"{0}\", \"client_secret\": \"{1}\", \"grant_type\": \"client_credentials\"", configSection.ClientId, configSection.ClientSecret);
```

Comment:
This code works, but I would like it to be more readable. Instead of explicitly creating the JSON payload, I would create a new class to model the payload and use Newtonsoft JSON's JsonSerializerSettings to ignore the default values - in our case - "" (the empty string). [More details](https://www.newtonsoft.com/json/help/html/T_Newtonsoft_Json_DefaultValueHandling.htm)

Reply:
@sfdrogojan i thought about it but then its a third party dependency we have to add. But adding a dependency just to solve this once purpose didn't feel right to me. Thoughts?

Reply:
Newtonsoft.Json is already a dependency for the C# SDK. So it's just a matter of using it. No need to add a new dependency in the project.